winlogbeat/eventlog/eventlog.go

```diff
@@ -15,214 +16,104 @@ var (
-func (el eventLog) Name() string {
-	return el.name
-}
+// Record represents a single event from the log.
+type Record struct {
+	API string // The event log API type used to read the record.
 
-func newEventLog(uncServerPath, eventLogName string) *eventLog {
-	el := &eventLog{
-		uncServerPath: uncServerPath,
-		name:          eventLogName,
-		handles: newHandleCache(eventLogName, queryEventMessageFiles,
-			freeLibrary),
-		logPrefix: fmt.Sprintf("EventLog[%s]", eventLogName),
-	}
-	return el
-}
+	EventLogName  string    // The name of the event log from which this record was read.
+	SourceName    string    // The source of the event log record (the application or service that logged the record).
+	ComputerName  string    // The name of the computer that generated the record.
+	RecordNumber  uint64    // The record number of the event log record.
+	EventID       uint32    // The event identifier. The value is specific to the source of the event.
+	Level         string    // The level or severity of the event.
+	Category      string    // The category for this event. The meaning of this value depends on the event source.
+	TimeGenerated time.Time // The timestamp when the record was generated.
+	User          *User     // The user that logged the record.
 
-func NewEventLoggingAPI(eventLogName string) EventLoggingAPI {
-	return newEventLog("", eventLogName)
+	Message        string   // The message from the event log.
+	MessageInserts []string // The raw message data logged by an application.
+	MessageErr     error    // The error that occurred while reading and formatting the message from the event log.
 }
 
-func NewRemoteEventLoggingAPI(uncServerPath, eventLogName string) EventLoggingAPI {
-	return newEventLog(uncServerPath, eventLogName)
+// String returns a string representation of Record.
+func (r Record) String() string {
+	return fmt.Sprintf("Record API[%s] EventLogName[%s] SourceName[%s] "+
+		"ComputerName[%s] RecordNumber[%d] EventID[%d] Level[%s] "+
+		"Category[%s] TimeGenerated[%s] User[%s] "+
+		"Message[%s] MessageInserts[%s] MessageErr[%s]", r.API,
+		r.EventLogName, r.SourceName, r.ComputerName, r.RecordNumber,
+		r.EventID, r.Level, r.Category, r.TimeGenerated, r.User,
+		r.Message, strings.Join(r.MessageInserts, ", "), r.MessageErr)
 }
 
-// LogRecord represents a single record from an event log.
-type LogRecord struct {
-	EventLogName  string
-	SourceName    string
-	ComputerName  string
-	RecordNumber  uint64
-	EventID       uint32
-	EventType     string
-	EventCategory string
-	TimeGenerated time.Time
-	UserSID       *SID
-	Message       string
-}
+// ToMapStr returns a new MapStr containing the data from this Record.
+func (r Record) ToMapStr() common.MapStr {
+	m := common.MapStr{
+		"@timestamp":   common.Time(r.TimeGenerated),
+		"eventLogName": r.EventLogName,
+		"sourceName":   r.SourceName,
+		"computerName": r.ComputerName,
+		// Use a string to represent this uint64 data because its value can
```

Comment:
Yes, we certainly could have this issue elsewhere (like this one possibly elastic/topbeat#152).